### PR TITLE
ENH - CLI installation messages for uninstalled solvers

### LINKED ISSUE #553: CLI - ``benchopt`` prints "already available" even for uninstalled solvers
# Description
When attempting to install a solver through CLI, ``benchopt`` may prints "already available" and thereby not install the solver despite the latter not being installed.

# Reproduce

Assuming that ``benchopt`` is already installed

```bash
git clone https://github.com/mathurinm/benchmark_lasso.git@approx
cd benchmark_lasso

benchopt install . -s apgcd
```

or from the ``benchmark_lasso`` repo

```bash
git remote add mathurinm https://github.com/mathurinm/benchmark_lasso.git
git fetch mathurinm approx
git checkout approx

benchopt install . -s apgcd
```

# Debugging

Investigation (thanks to @mathurinm) shows that this happens because the import of the [solver is done inside the ``run`` function](https://github.com/benchopt/benchmark_lasso/pull/63/files#diff-2b2acee23dfb466767ea8234aef14182963cbde28c8400b341d2c76e3e9f7e6fR30) instead of the safe import context. Therefore it is not a bug in ``benchopt``.

# Suggestion

It would much better to elucidate this to the user, for instance by printing a message to hint at putting all imports in safe import context.

### PR DESCRIPTION
**Pull request by Badr-MOUFAD**
This close #553 by clarifying skips of classes (Objective, Solver, Dataset) installation.

Preview:

Screenshot from 2023-05-16 14-39-20

### CODE REVIEW

File: benchopt/utils/dependencies_mixin.py [hunk 1]
```diff
@@ -73,56 +71,6 @@ def is_installed(cls, env_name=None, raise_on_not_installed=None,
                 env_name=env_name, raise_on_error=raise_on_not_installed
             ) == 0
 
-    @classmethod
-    def install(cls, env_name=None, force=False):
```
Comment (Badr-MOUFAD) [inline benchopt/utils/dependencies_mixin.py:77]: While working on enhancing the installation message, I discovered that this method serves the same purpose as [``collect``](https://github.com/benchopt/benchopt/blob/69a62c2b6dd63d944229d0251ec0f4de1e24b93a/benchopt/utils/dependencies_mixin.py#L127) method, yet it is never used.

Perhaps we can delete it?

Comment (Badr-MOUFAD) [reply benchopt/utils/dependencies_mixin.py]: My bad! The CI pointed me to where it is used.

**Comment by mathurinm** [comment]
Thanks for the PR, I have 2 questions:
- as is, is it clear enough for the average user (who may not know what a safe import context is)?
- on a related note, if there is no safe import context in the Solver/Dataset, does it make sense to print this message, shouldn't we print something else?

**Review by tomMoral — APPROVED** [review]
LGTM

File: benchopt/utils/dependencies_mixin.py [hunk 2]
```diff
@@ -159,7 +161,11 @@ def collect(cls, env_name=None, force=False):
             post_install_hooks = [cls._post_install_hook]
         else:
             env_suffix = f" in '{env_name}'" if env_name else ''
-            print(f"- '{cls.name}' already available{env_suffix}")
+            colored_cls_name = colorify(f'{cls.name}', YELLOW)
+            print(
+                f"- {colored_cls_name} already available{env_suffix}\n"
+                "  All imports within `safe_import_context` are installed."
```
Comment (tomMoral) [inline benchopt/utils/dependencies_mixin.py:167]: RE-reading the comment from @mathurinm, maybe this is clearer, less `benchopt` jargon:
```suggestion
                f"  No ImportError raised from {cls.module_filename}."
```

Comment (Badr-MOUFAD) [reply benchopt/utils/dependencies_mixin.py]: The CI fails.
From the error logs, it appears that not all objects have ``module_filename``
```bash
=========================== short test summary info ============================
FAILED benchopt/tests/test_cli.py::TestInstallCmd::test_benchopt_install - AttributeError: type object 'Objective' has no attribute 'module_filename'
FAILED benchopt/tests/test_cli.py::TestInstallCmd::test_benchopt_install_in_env - AttributeError: type object 'Objective' has no attribute 'module_filename'
FAILED benchopt/tests/test_cli.py::TestInstallCmd::test_benchopt_install_in_env_with_requirements - AttributeError: type object 'Solver' has no attribute 'module_filename'
======= 3 failed, 139 passed, 6 skipped, 7 warnings in 194.93s (0:03:14) =======
Error: Process completed with exit code 1.
```

**Comment by tomMoral** [comment]
There is a missing `_`

File: benchopt/utils/dependencies_mixin.py [hunk 3]
```diff
@@ -159,7 +161,11 @@ def collect(cls, env_name=None, force=False):
             post_install_hooks = [cls._post_install_hook]
         else:
             env_suffix = f" in '{env_name}'" if env_name else ''
-            print(f"- '{cls.name}' already available{env_suffix}")
+            colored_cls_name = colorify(f'{cls.name}', YELLOW)
+            print(
+                f"- {colored_cls_name} already available{env_suffix}\n"
+                f"  No ImportError raised from {cls.module_filename}."
```
Comment (tomMoral) [inline benchopt/utils/dependencies_mixin.py:167]:
```suggestion
                f"  No ImportError raised from {cls._module_filename}."
```

**Pull request merged**